### DISCUSSION TITLE
Smart tags discussion thread

### DISCUSSION THREAD

**Discussion by atmoio** [standardnotes/forum#2513] [Show and tell]
Creating a dedicated thread for any questions related to [Smart Tags](https://standardnotes.com/help/42/how-do-i-view-a-list-of-untagged-notes-and-create-other-dynamic-filters). Feel free to share any tags you've created, or ask for help on how to construct a custom query.

**Comment by onlinemumie** [comment]
I found some JavaScripts references on [w3schools](https://www.w3schools.com/jsref/jsref_obj_string.asp) and modified one of the code examples on the [Smart Tag](https://standardnotes.org/help/42/how-do-i-view-a-list-of-untagged-notes) help page.

Now I've got a working Smart Tag for finding dates in titles. I always write dates within parenthesis, and this one will find any note that has _(20_ in the title, for example _(2018-07-28)_ or _(2020-10-01)_.

`!["Date", "title", "includes", "(20"]`

The string reference "includes" can be used to find text strings in the note text as well. Just replace "title" with "text". For example

`!["Buy", "text", "includes", "=Buy"]`

to find any note with _=Buy_ in the text.

**Comment by atmoio** [comment]
Looks good! More advanced, but you can use Regex too to do more detailed date finding (because your solution won't work once we enter the year 3000 ;)

**Comment by onlinemumie** [comment]
Sounds interesting, Unfortunately, I'm not a coder and know nothing of Javascript syntax. Care to give an example?

I also wonder if it's possible to combine "search terms" in a smart tag, for example to find all notes that contains "=Buy" and "+Garage", or all notes that contains "=Buy" and have the tag "Home".

**Comment by atmoio** [comment]
It's not currently possible to combine terms, but that may be an interesting addition.

Ah I forgot to include a regex example in the help post. Will add this later:

Regex is also supported. But be sure to escape your / and \ as necessary.

```
!["Only Letters", "title", "matches", "^[a-zA-Z]+$"]
```

**Comment by onlinemumie** [comment]
As I said, I'm not a coder. I understand that your example returns all notes with only letters, but I have a hard time figuring out how to modify the code for date formats and such. Can you recommend some good links on how to write the specific syntax for these kind of queries? What I've found so far on the web doesen't work and looks very different from your example.

**Comment by atmoio** [comment]
Your best bet for regex is an online builder:

https://regex101.com/
http://rubular.com/

It's definitely not easy, but very powerful.

**Comment by onlinemumie** [comment]
Perhaps not rocket science, but after some research I found (and tweaked) some regex code that will recognize the yyyy-mm-dd date format in titles. It won't check for valid dates though.

`!["Date", "title", "matches", "\\d{4}-\\d{2}-\\d{2}"]`

**Comment by atmoio** [comment]
Nice!

**Comment by onlinemumie** [comment]
I found some code for making negative searches at [stackoverflow.com](https://stackoverflow.com/questions/1538512/how-can-i-invert-a-regular-expression-in-javascript). Perhaps someone will find it useful.

`!["Not in text", "text", "matches", "^(?!.*(word1|word2|word3))"]`

This tag will search for notes NOT containing word1, word2, or word3.

It will only work for words on the first line of text, but that can probably be changed if .* is replaced or combined with some other token such as `\s` (whitespace character). For a list of tokens, check out the links provided by mobitar above. I can also recommend [Regular-Expressions.info](https://www.regular-expressions.info/).

It is also possible to use symbols such as + and * as part of words, but `\\` must be included before the symbol to escape its function.

Personally, I use _* or _+ on the first line of text to identify notes as part of themes or projects, so the smart tag I use (which will look for notes NOT marked as _+projects or _*themes) looks like this:

`!["Unmarked", "text", "matches", "^(?!.*(_\\*|_\\+))"]`

**Comment by ionisingTuna** [comment]
Nice one guys, please consider this one, which will check for numbered lists:
`^\d\.`  You can find it [here](https://regex101.com/r/scqbRf/2).
I have tried to convert it into the Standard Notes format unsuccessfully so far:
`!["Numbered List", "text", "includes", "^\\d(\.)"]`

Could one of you please show me the way? Cheers.

**Comment by onlinemumie** [comment]
Hello, ionisingTuna. I tried the following for your particular problem:
`!["Numbered List", "text", "matches", "\\d\\.\\s"]`
And it found any note with a digit followed by a dot and a whitespace character. Not very useful for numbered lists.
After some testing, I ended up with the following code. It found all my numbered lists, because all my numbered lists start with a 1 followed by a dot and a whitespace character:
`!["Numbered List", "text", "matches", "1\\.\\s"]`
Perhaps not perfect, but hey, it works!

**Comment by ionisingTuna** [comment]
_Thanks a million._ This works perfectly.

**Comment by atmoio** [comment]
_From a user email interaction:_

I have a tag called "Journal" that I'd like to create a view for.

I've tried these 3 options with no success.

```
- !["Journal", "tags", "=", "Journal"]
- !["Journal", "tags.title", "=", "Journal"]
- !["Journal", "title", "=", "Journal"]
```

I just want my 'Journal' tag to show up on the top under a "View".

**Solution:**

```
!["Journal", "tags", "includes", ["title", "=", "Journal"]]
```

**Comment by jacolind** [comment]
Can I do text includes "todo" or "TODO" and tag includes "project1"?

**Comment by atmoio** [comment]
Try

```
!["Todo", "text", "includes", "todo"]
```

As for a single predicate that satisfies all 3 conditions, I'm not sure. I'd have to review the code to see if that's possible.

**Comment by neaxi** [comment]
Is there any way to use Smart Tags in the Android app to display untagged notes? In similar way web app utilizes `tags.length = 0`?

**Comment by atmoio** [comment]
No, smart tags aren't currently supported on mobile.

Update: Smart tags are now available on mobile. On mobile, they can be viewed and used, but not created. Only way to create smart tags is with the Folders component.

**Comment by nickAtWybar** [comment]
I've found the default 'last day, last week' very useful views. Extending that, I'm trying to figure out what the comparison operator would be for 'last hour'.

['Last Hour','updated_at',*comparison operator. updated_at is less than hours time delta*,2]

Anybody have thoughts on how to do this?

Thanks.

**Comment by atmoio** [comment]
It should just be

```
!["Last Hour", "updated_at", ">", "1.hours.ago"]
```

**Comment by atmoio** [comment]
Also, smart tags are now supported in the mobile 3.0 beta: https://twitter.com/StandardNotes/status/1085185273116483584

**Comment by quartertone** [comment]
Is there a way to gather a list of notes that have attached files (via FileSafe).
Along those same lines, is there a way to identify notes that are using ___x___ extension? For example, if I want to see all my task lists in one place, or all my code work, etc.

**Comment by atmoio** [comment]
> Is there a way to gather a list of notes that have attached files (via FileSafe).

I don't think so. Under the hood, files have associations on notes, but notes don't have associations on files. So the notes don't know they have files.

> is there a way to identify notes that are using x extension?

Not automatically. For tasks, you'd know that the note has to start with `- []`, so that could be a good filter.

**Comment by onlinemumie** [comment]
Got this to work for finding lists with both finished and unfinished tasks ...
`!["Lists", "text", "startsWith" , "- ["]`

**Comment by agba55** [comment]
Non-tech here... i have several project tags Project1.todo, Project2.todo, Project3.todo.

How could i search for all tags *.todo ... 

(I tried several regex without success... also endsWith didn't give any results back..)

**Comment by atmoio** [comment]
@agba55 try this:

```
!["Todo", "tags", "includes", ["title", "includes", ".todo"]]
```

**Comment by agba55** [comment]
> @agba55 try this:
> 
> ```
> !["Todo", "tags", "includes", ["title", "includes", ".todo"]]
> ```

Oh wow that worked. So simple... thanks!!!!!

**Comment by Habmala** [comment]
Hi.

I'm wondering if there is a list of comparison operators. I specifically want to exclude a tag from a view so that I can use the program to write my blog posts in markdown but not have to see them with all my other notes. But I cant seem to find what to write to exclude a tag instead of including it.

Thanks.

**Comment by atmoio** [comment]
Hmm..there is no such operator, although there definitely should be. I'll make note of this for a future update. I think the closest you might get to accomplishing what you want would be a regex based filter. You can name your blog post tag starting with some special character, like _blog, then have the filter look for notes that have tags that don't start with that special character.

Something like:

```
!["Not Blog", "tags", "includes", ["title", "matches", "...regex..."]]
```

**Comment by Habmala** [comment]
Hi, thanks for the reply. I got it (kind of) working using this:
`
!["No blog", "tags", "includes", ["title", "matches", "^((?!Blogpost).)*$"]]
`
It gives any note that has tags that are not "Blogpost", however it also excludes untagged notes. I tried to find a work around, but have a hard time thinking of a way to match null. I don't really have any untagged notes outside of testing this, so I'm giving up for now, but if anyone has any idea I'd love to hear.

Thanks again.

**Comment by chegeiser** [comment]
Trying to get a Smart Tag to find all notes containing dates that follow the todotxt format: due:yyyy-mm-dd

This works: `!["Dates", "text", "matches", "due:\\d{4}-\\d{2}-\\d{2}"]`

Can also adjust it to this: `!["Dates", "text", "matches", "due:\\d{4}-\\d{1,2}-\\d{1,2}"]` which isn't exactly todotxt format but might be handy here.

**Comment by naveenroy** [comment]
@chegeiser, I'm a heavy todotxt user as well (I use Simpletask on Android). Just curious - are you able to completely transition to SN for your todotxt requirements? If yes, what is your process flow like, using Smart tags?

**Comment by chegeiser** [comment]
@naveenroy This is definitely not a replacement for anything like Simpletask or another todotxt app/software. But can help if you bury occasional due dates in your notes. What I'd really love is a todotxt extension for SN that pulls all todotxt formatted notes into one page, with a reference or link to the SN they come from. :-) Unfortunately I'm not proficient enough to create this myself.

**Comment by atmoio** [comment]
A view/smart tag for getting all Protected notes:

```
!["Protected", "content.protected", "=", true]
```

**Comment by christianhans** [comment]
Is there a way to create a view that shows notes including TODO in their title but also shows notes with TODO inside the note's text?

**Comment by atmoio** [comment]
Compound predicates are now possible, so something like this should work:

For notes whose title **and** text contain TODO:
```
!["Todo", "ignored", "and", [["content.title", "includes", "TODO"],  ["content.text", "includes", "TODO"]]]
```

For notes whose title **or** text contain TODO:
```
!["Todo", "ignored", "or", [["content.title", "includes", "TODO"],  ["content.text", "includes", "TODO"]]]
```

("ignored" just represents fields that are ignored for compound predicates, so any value can be put there.)

**Comment by damphi** [comment]
About compound predicates/opertors !
Suppose I am sharing all my notes with my wife, but I would like my wife to see on her smartphone all notes with tag “wife” or “bank” or “book”
!["Wife", "ignored", "or", [["tag", "includes", "wife"],  ["tag", "includes", "bank"],  ["tag", "includes", "book"]]]

**Comment by notaustil** [comment]
I wanted to know what I went through last year at this time of the year.
```
!["Same period last year", "created_at", "and", [["created_at", "<", "365.days.ago"], ["created_at", ">", "395.days.ago"]]]
```

**Comment by rjkantor** [comment]
If I want to modify the Untagged View to exclude items in the Trash folder, how do I exclude items in the Trash folder?

`!["Untagged", "tags.length", "=", 0]`

Robb

**Comment by atmoio** [comment]
@rjkantor try a compound `and` predicate:

```
!["Untagged", "ignored", "and", [["tags.length", "=", 0], ["content.trashed", "=", false]]]
```

**Comment by rjkantor** [comment]
It seems when I look at untagged or the untagged and ignore trashed a ghost record is added to the tagged folder.  It seems to be there always.  Does anyone else see this record? I delete it, and it comes back when I go back to that view.  

Thank you,
  Robb

**Comment by atmoio** [comment]
Hmm I'm not seeing this. Can you share a screenshot of what you mean?

**Comment by jmason888** [comment]
Is there a possibility of a "not" predicate, for cases where the query is not easily reversible?

What I mean is, if I want untagged or unprotected notes, it is easy enough to compare a flag to false rather than true.  But if I want something like "notes that _are not_ tagged with any of the following tags", it would be much easier to put a "not" as the outermost predicate, rather than pushing the not through the "or" using De Morgan's rule, then reverse all the regexes with negative lookaheads...

**Comment by atmoio** [comment]
Hmm there's no way to do that currently, but agreed it would be useful. Perhaps one way would be to introduce a special syntax in the operator fragment.

`!["Foo Notes", "title", "startsWith", "Foo"]` 
can be:
`!["Foo Notes", "title", ":not:startsWith", "Foo"]`

Here we theoretically add `:not:` as a prefix in the operator. Are there any other commands besides "not" you think would be useful?

**Comment by quartertone** [comment]
Is there a way to include archived notes in the search parameters? (Eg, filtering for notes with specific title both archived and un-archived.) I would like to have notes show up in the smart folders but not in the main "All" notes section.

**Comment by atmoio** [comment]
`archived` is a top level property, so you can just do `!["Archived", "archived", "=", true]`, or use compound predicates to combine it with others.

**Comment by quartertone** [comment]
I tried `!["include archived", "ignored", "and", [["title", "matches", "^[dD]emo"], ["archived", "=", true]]]` but it didn't match anything. So I tried `!["Archived", "archived", "=", true]` which didn't match anything either.

**Comment by atmoio** [comment]
Oh you know what. Archived and trashed are special system cases now. You can't create filters using those. I'll update the docs. What you'll need to do if you want this to work (on desktop only probably) is select Options in the list of notes panel, and choose to display "Archived Notes".

**Comment by onlinemumie** [comment]
Always wanted to combine tag + title/text search in Standard Notes. With the new combined predicates it's possible. Perhaps someone will find this example useful (it will search for notes with the "work" tag and "Prio" in the title):
`!["Prio work", "ignored", "and", [["tags", "includes", ["title", "startsWith", "work"]], ["title", "includes", "Prio"]]]`

**Comment by quartertone** [comment]
> What you'll need to do if you want this to work (on desktop only probably) is select Options in the list of notes panel, and choose to display "Archived Notes".

This doesn't really accomplish what I'm looking for. I want to be able to hide a note from the main ("All notes") list, but have it show up in the Smart tags folder that corresponds to the search criteria. If I choose to display archived notes, the archived note appears in All Notes again and it's as if I didn't do anything (other than the "Archived" flag showing up).

**Comment by atmoio** [comment]
I don't believe there's a way to do that currently.

**Comment by scdssn** [comment]
I have a tag called productivity.today where each note title is formatted like "21 July, Sun", "22 July, Mon", etc. How could I construct a smart tag to get all notes from productivity.today that are newer than 7 days? Can't seem to figure it out. Thanks.

**Comment by scdssn** [comment]
Nevermind.
`!["Today New", "ignored", "and", [["tags", "includes", ["title", "startsWith", "productivity.today19"]], ["created_at", ">", "7.days.ago"]]]`

**Comment by csk** [comment]
I just wanted to share a filter it took me a while to figure out how to build.
The filter is **for notes having at the same time two tags**. For example, a note tagged as "personal" and "project".

`!["My Personal Projects", "ignored", "and", [["tags", "includes", ["title", "=", "personal"]], ["tags", "includes", ["title", "=", "project"]] ]]`

**Comment by ianjohnsa** [comment]
I have the most mundane question of all time - how do you actually delete a smart tag after having created it?

**Comment by johnny243** [comment]
Hey there @ianjohnsa,

You can delete a Smart Tag by clicking on the red button next to it:

![image](https://user-images.githubusercontent.com/5891646/85935919-d337bb80-b8c3-11ea-95a5-4ab3e99350ea.png)

---

You can also use Batch Manager extension to delete your Smart Tags. Simply open Batch Manager, click `Smart Tags`, then select the items that you want to delete. After, click the delete button.

![image](https://user-images.githubusercontent.com/5891646/85936023-cc5d7880-b8c4-11ea-89b5-4f12982ec3d2.png)

Hope this helps!

**Comment by lissy93** [comment]
This feature is awesome, yet another reason I love Standard Notes! 🙌 

Quick question; Does anyone know if Smart Tags are compatible with folders? In the [Folders documentation](https://standardnotes.org/extensions/folders) it does say: *"Folders also supports Smart Tags, which allow you to build custom filters for viewing your notes."*

Bit I couldn't figure out how to do it. I tried adding a dot in the Smart Tag name, to create a sub-folder, (like this, `!["Projects.My Project Name", "title", "includes", "Demo-Project"]`), and I tried to click and drag, neither worked. Maybe it's not possible, in which case no big deal. Just thought I would ask the experts, as if it is doable, it would be really neat!

Thanks in advance, 😊
Alicia

**Comment by atmoio** [comment]
Smart tags are indeed possible in Folders, but you cannot nest smart tags under other tags, or move them in any sort of fashion. They are single-level entities and only exist in your "Views" section.

**Comment by lissy93** [comment]
Thanks for the clarification @mobitar, makes sense - shame it's not possible, as I have a ton of smart tags and I'd of liked to be able to better organize them, but it's no big deal 😃

**Comment by joshcsullivan** [comment]
I have an odd smart tag filter I'm trying to figure out how to build, so hoping I might get some help on here.

I use the Folders extension, and have tags that start with `Personal` or `Work`, unfortunately due to some differences in how tags are inherited, there are some notes below each that do not get the parent `Personal` or `Work` (respectively). I'm trying to build a query that shows notes tagged with the child tags only and not the Parent tag.

Here is an example structure of my tags:
```
Personal
    Blog
    Recipes
Work
    General
    Processes
```

The query would need to find any note tagged with General but does not have the tag Work.

In a perfect world, this would be the query:
```
!["Not Work", "tags", "includes", [["title", "isNOT", "Work"], ["title", "startsWith", "Work"]]]
```
The way I read that is it does not find tags that are exactly `Work`, but does find tags that start with `Work`.

Hoping v4 will bring `isNot` or `doesNotEqual` to filters, but in the mean time, any help trying to get this query working would be appreciated.

**Comment by shompoe** [comment]
Hi,

I don't think you can do this at the moment.

If you think about the syntax...   "tags", "includes", ...    Then you are effectively saying "give me notes that include xyz tag" or in your case "give me notes that include any tag that is not xyz". But a note with abc tag and xyz tag is still a note with a tag that is not xyz.

 I was able to confirm this using regex:
  
!["Not Work", "tags", "includes", ["title", "matches", "^(?!.*Work).*$"]]

Which matches all notes not tagged 'Work', but with the caveat that even if a note is tagged 'Work' it is still matched if it has another tag that is not 'Work'. Pretty useless right?

--- 
Mark Desrousseaux



Nov 10, 2020, 13:37 by notifications@github.com:

**Comment by JaspalSuri** [comment]
Hi @joshcsullivan,

> Hoping v4 will bring `isNot` or `doesNotEqual` to filters, but in the mean time, any help trying to get this query working would be appreciated.

This currently isn't on our roadmap, but I've put in a request for this!

Thank you @shompoe for your help! I can confirm that it isn't currently possible.

**Comment by clough-fyi** [comment]
Hi all - any chance there's a smart tag for looking at all listed notes? Ie all notes that have been published?

It would be good to have one for all publicly published - so all listed notes - and one for all privately published to keep track of shared notes.

Is this possible?

**Comment by JaspalSuri** [comment]
Hi @clough-fyi,

I'm afraid that there currently isn't a [Smart Tag or Dynamic filter](https://standardnotes.org/help/42/how-do-i-view-a-list-of-untagged-notes-and-create-other-dynamic-filters) that would return a list of notes that were published to Listed. You would have to either (a) tag notes as you publish them with something like `Listed.Published` and/or (b) view the list of posts (both public and private) published via opening any note > clicking on **Actions** > **Listed** | **Settings**, and scroll down to **All Posts**. 

The section currently sorts posts from oldest to newest; while it includes both public and private notes in a single list it labels each one accordingly.

**Comment by hlascelles** [comment]
I was spending some time trying to wrangle some nested arrays when it occurred to me that whitespace (including linebreaks) is not important.

ie, if it makes your life easier, you can keep your smart tags "pretty printed", and when adding them just paste them in like this, and they will work just fine:

```
![
  "My Filter",
  "ignored",
  "and",
  [
    [
      "tags",
      "includes",
      ["title", "=", "todo"]
    ],
    [
      "title",
      "matches",
      "This Title|Some Prefix.*|\\d{4}-\\d{2}-\\d{2} .*"
    ]
  ]
]
```

**Comment by ChristophCan** [comment]
Hi, I am trying to set up a filter which shows me notes from the current week, or last week. I don´t want to see tasks that are within the last 7 days, but rather showing all tasks I am working on this current work week (Monday through Sunday). 

I tried:
`!["CurrentWeek", "updated_at", ">", "1.weeks.ago"]`
which only shows an empty list.

Is there such a thing as `1.weeks.ago`?

Thank you!

**Comment by atmoio** [comment]
Currently it only accepts "days" and "hours", not weeks. So try "7.days.ago". Also might need to be `< 7 days ago` rather than `> 7 days ago`.

**Comment by ChristophCan** [comment]
Thank you for your help! I will then try that.

So there is currently no option for searching "within current week"? 
For example if it´s Wednesday, I want wo see all tasks I edited on Monday, Tuesday, Wednesday.
If it´s Friday I want to see all tasks from Monday, Tuesday, Wednesday, Thursday, Friday...

**Comment by atmoio** [comment]
No there is no such option.

**Comment by gogolim** [comment]
Hi everyone. I was wondering if it's possible to create nested / combined filters?
For example, I would like to create a filter which shows all notes with a "Monthly" tag and the "updated_at" older than a month AND all notes with a "Weekly" tag and "updated_at" older than a week in the same view.
The idea is to have notes to review, but some on a weekly basis and some on a monthly basis.

As far as I understand the code, that would have to look something like this:
```
![
  "To Review",
  "ignored",
  "or",
[
  ["ignored","ignored","and",
    [
      "tags",
      "includes",
      ["title", "startsWith", "Monthly"]
    ],
    [
      "updated_at",
      "<",
      "30.days.ago"
    ]
  ],
	["ignored","ignored","and",
 [
      "tags",
      "includes",
      ["title", "startsWith", "Weekly"]
    ],
    [
      "updated_at",
      "<",
      "7.days.ago"
    ]
]
]
]
```

However, this code shows an empty filter, even though the single filters for weekly and monthly (tags & updated_at) show the correct notes. It's the combination that does not work.
Did I make a mistake in the code or is it simply not possible to combine two filters?

**Comment by JaspalSuri** [comment]
Hi @gogolim, I don't think it's possible but I'll double-check with the dev team on this.

**Comment by atmoio** [comment]
Compound predicates are supported, take a look at a test case here: https://github.com/standardnotes/snjs/blob/master/packages/snjs/test/lib/predicate.test.js#L260

**Comment by gardener0** [comment]
I could use help looking for all notes that have two tags apply.  For example, I want to see everything tagged both "call" and "doToday" so I can see the list of calls I'd like to make today.  The following is as far as I've gotten.

!["call today", "ignored", "and", ["tags", "includes", ["title", "=", "call"]], ["tags", "includes", ["title", "=", "doToday"]]]

**Comment by atmoio** [comment]
You'd want it to look like this:

```json
[
  "call today",
  "ignored",
  "and",
  [
    ["tags", "includes", ["title", "=", "call"]],
    ["tags", "includes", ["title", "=", "doToday"]]
  ]
]
```

Flattened that would be:

```json
["call today", "ignored", "and", [["tags", "includes", ["title", "=", "call"]],["tags", "includes", ["title", "=", "doToday"]]]]

```

**Comment by SvartrShell** [comment]
Hi - I'm hitting a bit of snag with this and I've had no luck with regex queries.

i have 2 main parent tags (folders?) Called "Work" and "Personal" - Each have several subfolders = Such as "Personal.Example" or "Work.Example2"

I like to keep things specific so nothing is tagged just "work" and all notes have more specific subtags applied. Nothing shows when looking under the work tag - I would like to make a smart tag that shows all notes with tags starting with "work"

Ive tried a fair amount but nothing seems to work - At one point I had a regex running but it also picked up stuff tagged "network" which weren't work specific
A few simple ones ive tried
!["Work", "tags", "includes", ["title", "in", ["Work"]]]
!["Work", "tags", "includes", ["title", "startwith", ["Work"]]]

(Theres been more but ive lost track now!)

Is what I'm trying to acheive here even possible?

**Comment by atmoio** [comment]
With the new hierarchical folders structure, sub tags do not contain the name of their parent. So Work.Example will just be called Example, but it will have a parentId reference to the Work via its UUID. So you can't filter using name.

However if you enable Preferences > Add all parent tags when adding a nested tag to a note (which is enabled by default), then adding Work.Example to a note will also add Work. In which case the filters will work.

Last option would be to create a filter based on the parentId of the tag. But I haven't really tested this before. Something like `tags includes parentId = workTag.uuid` (pseudocode)

**Comment by SvartrShell** [comment]
> 

Thanks for this, how would i find a tags UUID?

**Comment by atmoio** [comment]
Probably the easiest (but somewhat cumbersome) way to do this would be to export a decrypted backup file of your account data, search for the tag name, and find its uuid. We'll work on adding the tag uuid to the right click context menu, similar to what we do for notes.

**Comment by EweSparky** [comment]
How do I create a Smart View for a tag using the `Add Smart View` screen in v3.120.2? I’m trying to filter by `#parent-tag` while excluding `#parent-tag.child-tag`. The property list in the Builder section doesn’t include a tag property.

Thanks.

**Comment by effieeee** [comment]
Hi @StuartMorrisAU, unfortunately the builder isn't able to make that particular smart view just yet, but the Custom JSON creator has a sample which should help you do this:

![image](https://user-images.githubusercontent.com/54989748/205941765-c1705711-8a83-424f-beab-2ad87a7e71ee.png)

You just need to replace the names in the **value** property with your desired tag names!
